Docs home page and navigation refactor (#8807)
PrefectHQ/prefect: docs/index.md

---
icon: material/star-shooting-outline
title: Welcome to Prefect
description: Get started with Prefect, the easiest way to orchestrate and observe your data pipelines
tags:
    - getting started
    - quick start
    - overview
hide:
  - toc
---

# **Welcome to Prefect**

> ### The easiest way to orchestrate and observe your Python code { .main-page-subheader }

Reviewer: the phrase "observe your Python code" starts to sound too much like a tracer, and isn't really the unit of observation we're most focused on

Prefect enables you to build resilient data workflows so that you can understand, react to, and recover from unexpected changes. Bring your Python code, sprinkle in a few decorators, and go!

> With Prefect you gain easy:

Reviewer: ```suggestion
With Prefect you gain simple:
```

<figcaption>Prefect UI</figcaption>
</figure>

#### New to Prefect?
If you're ready to dive in and learn Prefect, try the [tutorial](/tutorial/index/) where you'll create a full data project, jump into Prefect [concepts](/concepts/index/), or explore the [guides](guides/index/) for common use cases. <div style="height: 10px"></div>

[Tutorials](/tutorials/){ .md-button .md-button--primary .main-button--primary .full}  [Concepts](/concepts){ .md-button .main-button--secondary .full }  [Guides](guides){ .md-button .main-button--secondary .full }

<div style="height: 10px"></div>
<p>Or, read on for a quick sample of Prefect.</p>
---

## Quick Start: Hello Prefect

Install Prefect with 

<div class="terminal">
```bash
pip install -U prefect
```
</div>

See the [install guide](/getting-started/installation/) for more detailed instructions.

### Run a basic flow

Import `flow` and decorate your Python function using the [`@flow`][prefect.flows.flow] decorator.

```python hl_lines="1 3"
from prefect import flow

@flow
def my_favorite_function():
    print("What is your favorite number?")
    return 42

print(my_favorite_function())
```

> Thats it! Your function is now a flow. Run the code as you normally would, and you'll see its orchestration log:

Reviewer: ```suggestion
Thats it! Your function is now a flow. Run the code as you normally would, and you'll see its execution via the Prefect logs:
```

<div class="terminal">
```bash
$ python hello_prefect.py
15:27:42.543 | INFO    | prefect.engine - Created flow run 'olive-poodle' for flow 'my-favorite-function'
15:27:42.543 | INFO    | Flow run 'olive-poodle' - Using task runner 'ConcurrentTaskRunner'
What is your favorite number?
15:27:42.652 | INFO    | Flow run 'olive-poodle' - Finished in state Completed()
42
```
</div>

> The Prefect orchestration engine observes the flow run state and saves it in a database. By adding a single decorator, you've gained some observability of your flow. There's much more you can do with Prefect, so keep rolling to learn how to add easy retries, notifications, scheduling and more!

Reviewer: ```suggestion
Prefect automatically persists all executions of your flow along with useful metadata such as start time, end time, and state. This is just the beginning, so keep exploring to learn how to add retries, notifications, scheduling and much more!
```

## Next Steps

> If you like what you see, try our [tutorials](/tutorials) and [guides](/guides), or read more about [Prefect concepts](/concepts).

Reviewer: ```suggestion
To learn more, try our [tutorials](/tutorials) and [guides](/guides), or go deeper with [Prefect concepts](/concepts).
```

---
icon: material/star-shooting-outline
title: Welcome to Prefect
description: Get started with Prefect, the easiest way to orchestrate and observe your data pipelines
tags:
    - getting started
    - quick start
    - overview
hide:
  - toc
---

# **Welcome to Prefect**
### The easiest way to orchestrate and observe your Python code { .main-page-subheader }

> Prefect enables you to build resilient data workflows so that you can understand, react to, and recover from unexpected changes.

Author: incorporated manually

> Bring your Python code, sprinkle in a few decorators, and go!

Author: incorporated by hand b/c outdated
